tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+const rule = require('../../../lib/rules/no-array-prototype-extensions');
+const RuleTester = require('eslint').RuleTester;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('@babel/eslint-parser'),
+  parserOptions: { ecmaVersion: 2020, sourceType: 'module' },
+});
+
+ruleTester.run('no-array-prototype-extensions', rule, {
+  valid: [
+    '[1, 2, 4].filter(el => { /* ... */ })',
+    'something.filter(el => { /* ... */ })',
+    "filterBy(something, 'foo')",
+    'this.filterBy()',
+    'something[0]',
+    'something[0].foo',
+    "get(something, 'foo.0.bar')",
+    "get(something, 'foo.0')",
+    "get(something, 'foo')[0]",
+    "get(something, 'firstObjects')",
+    "get(something, 'lastObjects')",
```

Comment:
To make this more clear about what you're trying to test:
```suggestion
    "get(something, 'notFirstObject')",
    "get(something, 'notLastObjet')",
```

Reply:
> Another note is that this looks like a candidate to be enabled as a `recommended` rule in the next major release (#1169).

Do you think the Deprecate array prototype extensions RFC (https://github.com/smilland/rfcs/pull/1) is a blocker to make this `recommended`? Also, could I remove the reference link for now and add it later once it's public?

Reply:
Good point, we can wait until the RFC is public and has gained support before making it `recommended`.

You can add the link once it's public.